superset-frontend/src/components/Select/Select.tsx

```diff
@@ -137,15 +143,16 @@ const Error = ({ error }: { error: string }) => (
 const Select = ({
   allowNewOptions = false,
   ariaLabel,
+  fetchOnlyOnSearch,
```

Comment:
What do you think of changing this prop to `fetchOnFocus`? I think it is more accurate as you are not typing anything but only focusing the input.

Reply:
> What do you think of changing this prop to `fetchOnFocus`? I think it is more accurate as you are not typing anything but only focusing the input.
>
> Also, I remember agreeing that we should just set this as the standard as it might be dangerous for performance otheriwse. What do you think?

I think you misunderstood the code. By default, the queries will only be triggered when the select is opened for the first time. This is controlled by the `loadingEnabled` state. 

The `fetchOnlyOnSearch` property is another behavior where the queries will only be triggered when there's a search input present.

Reply:
You are right. I was under the false impression this would only solve the lazy loading. Thanks for taking care of both cases!